### ISSUE TITLE
Can't receive more messages until I update the bot settings (into NR)

### ISSUE THREAD

**Issue by Barabba11** [windkh/node-red-contrib-telegrambot#287]
Hi, it's a bit strange, I have 14.8.7, running oon NR 1.3.5 and Node v14.17.2 on a Pi3b, usally it happens when the internet goes off, like a timeout in the polling create the problem after, I can't receive more messages on receiver node. The sender works ok. I should update the bot settings (without changing nothing), and after some seconds I'l receive on my alla answers from NR, for all messages I sent and that were't received.
So, after update, all the messages in queue exit from node and produce the result (I receive on phone the feedback). Is it maybe a problem connected to the old NR version?

**Comment by koosoli** [comment]
For me, this stopped working on the 18th of December. I think the last update was not good. I have the same constellation and I do not receive anymore messages.

**Comment by Barabba11** [comment]
have you tried to update the settings of module, about bot? I tolved me. By the way NR was always able to send messages, ony received mesages didn't go out from module, they were queued and after update I received all at ance

**Comment by koosoli** [comment]
Ok, I just saw that I am not on the latest version, but I can also not really update:

```
-----------------------------------------------------------
2022-12-30T12:54:32.771Z Install : node-red-contrib-telegrambot 14.8.7

2022-12-30T12:54:32.859Z npm install --no-audit --no-update-notifier --no-fund --save --save-prefix=~ --production --engine-strict node-red-contrib-telegrambot@14.8.7
2022-12-30T12:54:35.252Z [err] npm WARN config production Use `--omit=dev` instead.
```

What do I need to update in the settings? I am still using the same bot...hmmm

**Comment by Barabba11** [comment]
your device is not properly connected to internet, for this reason you've problems with Telegram too

**Comment by koosoli** [comment]
That is strange, Home Assistant is still connected, I did also deactivate adguard. But since it has been working, I did not really change anything.

**Comment by Barabba11** [comment]
it happened again, this time without an internet failure, send is ok, receive stop working until I update module settings

**Comment by windkh** [comment]
This seems to be the same as described here #282

**Comment by Barabba11** [comment]
> This seems to be the same as described here #282

In my opinion it doesn't look like that, The node have never had problems to send messages, any time, and it doesn suddenly. The problem is only about receving messages

**Comment by Barabba11** [comment]
today the commands were received and transmitted to nodes, the receiver node didn't produce output, then I updated the bot settings and it started working, but I haven't received the previous "queued" messages. A bit strange behaviour here

**Comment by windkh** [comment]
@Barabba11 
these type of problems were reported before. Everytime this only happens on a raspi. Don't ask me why.
in #282 it is described that it makes a difference how node-red was installed on the raspi.

Maybe you can try that or you can user the control node to find out of polling stalls when you discover those effects.

**Comment by Barabba11** [comment]
Thanks for kind support, 
I'm not the programmer.. but I would bet it's not depending on polling.. after all the commant are responsing now promptly, so the polling itself should working fine. At this time I can say I figured out 3 problems.
Not receiveng at all, commands or messages, but sending works fine. Update solves and got the queued messages
Receiving commands promptly and sending, but not receiving messages, updating solves but no queue
After command (which triggers Custom Keyboard) I can't get rid of it, the same flow was working fine with previous version, if I'm not wrong, was 10.0.4.

Can it be an incompatibility with NR or Nodejs? My version is not the latest.. I would like to downgrade back to 10.0.4, is it dangerous? Will it create problems or should I completely reconfigure bot after that? can you please kindly tell me how to force that downngrade? I suppose I shoould stop NR and use smething like npm i ... under the .node-red folder.
The telegram module was installed by palette, NR was installed under pi user, not root.
Thanks

**Comment by Barabba11** [comment]
May you kindly please tell me how may I downgrade back to 10.0.4 (or other version that you suggest) (should I uninstall first?) and if it's dangerous (I'll loose token or ay other problem) thank you!

**Comment by Barabba11** [comment]
I hope somebody can help me downgrade, thank you!

**Comment by windkh** [comment]
Just download the old package and copy it manually to the node_modules folder

**Comment by Barabba11** [comment]
thanks for reply, I followed your suggestion, I stopped NR, made a backup of existing files, overwritten with old release 10.4, started NR and got error
```
13 Jan 00:23:22 - [warn] ------------------------------------------------------
13 Jan 00:23:22 - [warn] [node-red-contrib-telegrambot/telegrambot] Error: Cannot find module 'socks5-https-client/lib/Agent'
```
Flow were stopped, so I replaced the backuped files and all seams ok. But I haven't downgraded, so how can I proceed?
If I?m not wrong there was something like npm i <name of old package", but I?m not sure and I'm afraid I could create problems/corruptions on my flow.
Thanks for help

**Comment by Barabba11** [comment]
I think the bug is somehow connected to the virtual keyboard, I've restarted NR and sent a normal text, for testing puroposes I've placed a receive node and created an echo, in the function I filter out any chance to move forward text contaning the keyboard respones and "/" commands. Other commands note I have are properly configured.
It looks like also a normal text triggers the "context.global.keyboard = {pending:true}", no sense, but same effect. SO I addecd in that function a context.global.keyboard = {pending:false}; and sent a text, I got echo, but I can't get rid of keyboard. 
Probably this thing in the long run creates the crash on receving messages

**Comment by Barabba11** [comment]
Hi, today I've upgraded all to the latest verisone (NR@3.02, nodeJS v14.21.2), I don't see  yet the update 14.9 avalable on palette, so I tried to force it with npm install node-red-contrib-telegrambot but it's not updraging. 
Now when I reboot NR I dont' see keyboard, but has I type a message (not a command) I see it, and I will not be out of it till a soft restart of module. Yhis is the same as before
So it's not due to old platform lik I hoped so to solve it, 
is it caused from the module or did I do someting wrong? Please help, than you

**Comment by windkh** [comment]
please send me your flow

**Comment by Barabba11** [comment]
Hi, thanks for your kins support, please have a look if something is wrong here 
`[{"id":"b469b56f.52bc7","type":"exec","z":"9cf3a6d.1104258","command":"vcgencmd measure_temp","addpay":false,"append":"","useSpawn":"","timer":"","name":"RPi Temp.","x":570,"y":1020,"wires":[["e6f8e3e0.08d87"],[],[]]},{"id":"5d205150.db2d1","type":"exec","z":"9cf3a6d.1104258","command":"top -d 0.5 -b -n2 | grep \"Cpu(s)\"|tail -n 1 | awk '{print $2 + $4}'","addpay":false,"append":"","useSpawn":"","timer":"","name":"CPU Load","x":570,"y":1100,"wires":[["f8d9657d.5ce198"],[],[]]},{"id":"3dc8ca11.8a3f7e","type":"exec","z":"9cf3a6d.1104258","command":"free | grep Mem | awk '{print $3/$2 * 100.0}'","addpay":false,"append":"","useSpawn":"","timer":"","name":"Free Memory","x":560,"y":960,"wires":[["db903535.d78d3"],[],[]]},{"id":"38a07b6c.0fe714","type":"exec","z":"9cf3a6d.1104258","command":"df -h","addpay":false,"append":"","useSpawn":"","timer":"","name":"Disk Usage","x":570,"y":1160,"wires":[["e73e4fb6.9bb7"],[],[]]},{"id":"4bec0b98.8cf43c","type":"inject","z":"9cf3a6d.1104258","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"*/10 0-23 * * *","once":true,"onceDelay":"300","topic":"","payload":"","payloadType":"date","x":110,"y":1080,"wires":[["de461dbf.3f3b98","5d205150.db2d1","3790321f.ed3486","9e559cf7.87eff8","a0714e85.5218d","68d54804.0d9be"]]},{"id":"e6f8e3e0.08d87","type":"function","z":"9cf3a6d.1104258","name":"","func":"str = msg.payload\nstringa = str.substring(5,9);\nrisultato=parseFloat(stringa);\nglobal.set(\"MonitorTemperatura\",risultato);\nmsg=null;\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":760,"y":1000,"wires":[[]]},{"id":"3a4569f0.35cabe","type":"tcp request","z":"9cf3a6d.1104258","name":"Test SSH port 22","server":"127.0.0.1","port":"22","out":"time","splitc":"100","x":550,"y":1360,"wires":[["226ff6c4.3ab44a"]]},{"id":"c61df51d.93b41","type":"tcp request","z":"9cf3a6d.1104258","name":"Test Telnet port 23","server":"127.0.0.1","port":"23","out":"time","splitc":"100","x":550,"y":1300,"wires":[[]]},{"id":"8b54ee2e.bc7338","type":"tcp request","z":"9cf3a6d.1104258","name":"Test port HTTP 80","server":"127.0.0.1","port":"1880","out":"time","splitc":"100","x":550,"y":1220,"wires":[["52a792d2.7633b4"]]},{"id":"8711b3cd.18b588","type":"telegram receiver","z":"9cf3a6d.1104258","name":"","bot":"cd6e647f.ad618","saveDataDir":"","filterCommands":false,"x":110,"y":1500,"wires":[["6d09c53.205373c"],[]]},{"id":"6d09c53.205373c","type":"function","z":"9cf3a6d.1104258","name":"NotificheTelegram tutto bene/internet","func":"\nif ((global.get(\"NotificheRete\")) & (msg.payload.content!=\"Attiva\") & (msg.payload.content!=\"Disattiva\") & (msg.payload.content[0]!=\"/\")) {\n    var messaggio=msg.payload.content;\n    var res = \"|Raspberry|: \"+messaggio;\n    context.global.keyboard = {pending:false};\n    msg = {payload:{chatId: 956528929, type:\"message\", content:res}}}\nelse {msg=null;}\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":1290,"y":1480,"wires":[["39f83657.5b0fd2"]]},{"id":"9a7d99af.56b3b","type":"timerswitch","z":"9cf3a6d.1104258","name":"Tutto bene 12:00","ontopic":"","offtopic":"","onpayload":"on","offpayload":"","disabled":false,"schedules":[{"on_h":"12","on_m":"00","on_s":"00","off_h":"12","off_m":"00","off_s":"05","valid":true}],"x":110,"y":1560,"wires":[["a99aab5b.b3fde8"]]},{"id":"a99aab5b.b3fde8","type":"function","z":"9cf3a6d.1104258","name":"Traduci per Telegram","func":"\nif (msg.payload==\"on\") {msg={payload:{content:\"Tutto bene!\"}};return msg;}","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":800,"y":1560,"wires":[["6d09c53.205373c"]]},{"id":"d6723f76.07b9a8","type":"ping","z":"9cf3a6d.1104258","protocol":"IPv4","mode":"timed","name":"Google-Ping","host":"8.8.4.4","timer":"600","inputs":0,"x":90,"y":1440,"wires":[["bb33a59a.cba3b"]]},{"id":"bb33a59a.cba3b","type":"function","z":"9cf3a6d.1104258","name":"Pinga-conta Telegram","func":"var internetnonfunzionava=global.get('internetnonfunzionava')||false;\nvar pingamem=global.get('internetconta')\nif (typeof pingamem == 'undefined') {pingamem=3;}\nmsg1=null;\n\nif (msg.payload===false) {pingamem--;\n    if (pingamem==0) { \n\t\tglobal.set('internetpinga', false); \n\t\tif (!internetnonfunzionava) {msg1={payload:{chatId: 956528929, type:\"message\", content:\"|Ping|: internet NON va\"}};global.set('internetnonfunzionava',true);}\n\t\tnode.status({fill:\"red\",shape:\"dot\",text:\"Internet non va:\"});}\t\t\n\telse {if (pingamem<0) {pingamem=0;}\n\tif (pingamem>0) {node.status({fill:\"yellow\",shape:\"dot\",text:\"|Ping|: \"+pingamem+\" perso\"});}}}\t\t\t\n\t\nelse {pingamem++;\n    if (pingamem==3) {\n\t\tglobal.set('internetpinga', true); \n\t\tif (internetnonfunzionava) {msg1={payload:{chatId: 956528929, type:\"message\", content:\"|Ping|: internet RIPRISTINATO\"}};global.set('internetnonfunzionava',false);}\n\t\tnode.status({fill:\"green\",shape:\"dot\",text:\"Internet Funziona\"});}\n\telse {if (pingamem>3) {pingamem=3;}\n\t\tif (pingamem<3) {node.status({fill:\"blue\",shape:\"dot\",text:\"|Ping|: \"+pingamem+\" in credscita:\"});}}}\n\nglobal.set('internetconta', pingamem);\nreturn msg1;\n","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":800,"y":1440,"wires":[["39f83657.5b0fd2"]]},{"id":"39f83657.5b0fd2","type":"telegram sender","z":"9cf3a6d.1104258","name":"","bot":"cd6e647f.ad618","haserroroutput":false,"outputs":1,"x":1610,"y":1260,"wires":[[]]},{"id":"f8d9657d.5ce198","type":"function","z":"9cf3a6d.1104258","name":"","func":"stringa = msg.payload\nrisultato=parseFloat(stringa);\nglobal.set(\"MonitorCPU\",risultato);\nmsg=null;\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":760,"y":1100,"wires":[[]]},{"id":"db903535.d78d3","type":"function","z":"9cf3a6d.1104258","name":"","func":"valore2=parseFloat(msg.payload);\nvalore=Math.round(valore2*10)/10;\nglobal.set(\"MonitorRAM\",valore);\nmsg=null;\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":760,"y":940,"wires":[[]]},{"id":"e73e4fb6.9bb7","type":"function","z":"9cf3a6d.1104258","name":"","func":"var re = /([0-9]{2})%/\nvar idx = msg.payload.search(re);\nvar str = msg.payload;\nif (idx >=0) {\n    str = msg.payload.substring(idx, idx + 2);\n}\nvar risultato = parseFloat(str);\nglobal.set(\"MonitorDisk\",risultato);\nmsg=null;\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":760,"y":1140,"wires":[[]]},{"id":"3a0c8c39.8e5f54","type":"function","z":"9cf3a6d.1104258","name":"Rileva soglie e formatta Telegram","func":"msg=null;\nif (global.get(\"NotificheSistema\")) {\ntemp=global.get(\"MonitorTemperatura\");\ncpu=global.get(\"MonitorCPU\");\nram=global.get(\"MonitorRAM\");\ndisk=global.get(\"MonitorDisk\");\nvar string=\"CPU:\"+cpu+\", Temp:\"+temp+\", RAM:\"+ram+\", Disk:\"+disk;\n\n\nif ((temp>57) | (cpu>35) | (ram>30) | (disk>30)){\n    node.status({fill:\"red\",shape:\"dot\",text:string});\n     msg={payload:{chatId: 956528929, type:\"message\", content:string}};}\nelse {node.status({fill:\"green\",shape:\"dot\",text:string});}}\n\nreturn msg;\n","outputs":1,"noerr":0,"initialize":"global.set('NotificheSistema',true);","finalize":"","libs":[],"x":1260,"y":1000,"wires":[["39f83657.5b0fd2"]]},{"id":"dfa5989a.cf1018","type":"function","z":"9cf3a6d.1104258","name":"Rileva risposte a connessioni","func":"msg=null;\nif (global.get(\"NotificheRete\")) {\nvar http=global.get(\"MonitorHTTP\") || false; global.set(\"MonitorHTTP\",false);\nvar telnet=global.get(\"MonitorTelnet\") || false; global.set(\"MonitorTelnet\",false);\nvar ssh=global.get(\"MonitorSSH\") || false; global.set(\"MonitorSSH\",false);\n\n\nvar string=\"HTTP:\"+http+\", Telnet:\"+telnet+\" SSH:\"+ssh;\n\nif ((http===false) | (ssh===false)) //(telnet===false) | per abbattere i log non lo includo\n\t{node.status({fill:\"red\",shape:\"dot\",text:string});\n\tmsg={payload:{chatId: 956528929, type:\"message\", content:string}};}\nelse {node.status({fill:\"green\",shape:\"dot\",text:string});}}\n\nreturn msg;","outputs":1,"noerr":0,"initialize":"global.set('NotificheRete',true);","finalize":"","libs":[],"x":1270,"y":1080,"wires":[["39f83657.5b0fd2"]]},{"id":"52a792d2.7633b4","type":"function","z":"9cf3a6d.1104258","name":"Verifica HTTP","func":"if ((msg.payload[0]==0x48) & (msg.payload[1]==0x54) & (msg.payload[2]==0x54)) {global.set(\"MonitorHTTP\",true); node.status({fill:\"green\",shape:\"dot\",text:\"HTTP OK\"});} \nelse {global.set(\"MonitorHTTP\",false);node.status({fill:\"red\",shape:\"dot\",text:\"HTTP KO\"});}\nreturn null;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":780,"y":1220,"wires":[[]]},{"id":"a3d3d7ea.210ed","type":"function","z":"9cf3a6d.1104258","name":"Verifica Telnet","func":"if ((msg.payload[0]==0xff) & (msg.payload[1]==0xfd) & (msg.payload[2]==0x18)) {global.set(\"MonitorTelnet\",true);node.status({fill:\"green\",shape:\"dot\",text:\"Telnet OK\"});\n} else {global.set(\"MonitorTelnet\",false);node.status({fill:\"red\",shape:\"dot\",text:\"Telnet KO\"});}\nreturn null;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":780,"y":1300,"wires":[[]]},{"id":"226ff6c4.3ab44a","type":"function","z":"9cf3a6d.1104258","name":"Verifica SSH","func":"if ((msg.payload[0]==0x53) & (msg.payload[1]==0x53) & (msg.payload[2]==0x48)) {global.set(\"MonitorSSH\",true);node.status({fill:\"green\",shape:\"dot\",text:\"SSH OK\"});} \nelse {global.set(\"MonitorSSH\",false);node.status({fill:\"red\",shape:\"dot\",text:\"SSH KO\"});}\nreturn null;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":770,"y":1360,"wires":[[]]},{"id":"de461dbf.3f3b98","type":"delay","z":"9cf3a6d.1104258","name":"","pauseType":"delay","timeout":"200","timeoutUnits":"milliseconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"outputs":1,"x":330,"y":1020,"wires":[["3dc8ca11.8a3f7e","b469b56f.52bc7"]]},{"id":"3790321f.ed3486","type":"delay","z":"9cf3a6d.1104258","name":"","pauseType":"delay","timeout":"100","timeoutUnits":"milliseconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"outputs":1,"x":330,"y":1300,"wires":[["8b54ee2e.bc7338","c61df51d.93b41","3a4569f0.35cabe"]]},{"id":"9e559cf7.87eff8","type":"delay","z":"9cf3a6d.1104258","name":"","pauseType":"delay","timeout":"150","timeoutUnits":"milliseconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"outputs":1,"x":350,"y":1160,"wires":[["38a07b6c.0fe714"]]},{"id":"e9585a35.b67ca","type":"inject","z":"9cf3a6d.1104258","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"60","crontab":"","once":true,"onceDelay":"30","topic":"","payload":"","payloadType":"date","x":120,"y":1200,"wires":[["5d205150.db2d1","a0714e85.5218d"]]},{"id":"68d54804.0d9be","type":"delay","z":"9cf3a6d.1104258","name":"","pauseType":"delay","timeout":"1","timeoutUnits":"seconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"outputs":1,"x":960,"y":1080,"wires":[["dfa5989a.cf1018"]]},{"id":"a0714e85.5218d","type":"delay","z":"9cf3a6d.1104258","name":"","pauseType":"delay","timeout":"1","timeoutUnits":"seconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"outputs":1,"x":960,"y":1000,"wires":[["3a0c8c39.8e5f54"]]},{"id":"16f89a0.5205be6","type":"telegram command","z":"9cf3a6d.1104258","name":"Abilita log sistema","command":"/sistema","description":"","registercommand":false,"language":"","scope":"default","bot":"cd6e647f.ad618","strict":false,"hasresponse":true,"useregex":false,"removeregexcommand":false,"outputs":2,"x":1040,"y":1180,"wires":[["49e8307f.b42dc8"],["b29392d1.f9f15"]]},{"id":"49e8307f.b42dc8","type":"function","z":"9cf3a6d.1104258","name":"Attiva?","func":"context.global.keyboard = {pending:true};\nvar opts = {reply_to_message_id: msg.payload.messageId, \n\treply_markup: JSON.stringify({keyboard: [ ['Attiva'], ['Disattiva']], 'resize_keyboard' : true,  'one_time_keyboard' : true })};\nmsg.payload.content = 'Attiva le notifiche per le risorse di sistema?';\nmsg.payload.options = opts;\nreturn [ msg ];","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":1240,"y":1140,"wires":[["39f83657.5b0fd2"]]},{"id":"b29392d1.f9f15","type":"function","z":"9cf3a6d.1104258","name":"Quindi esegui","func":"context.global.keyboard = {pending:false};\nif(msg.payload.content === 'Attiva') {global.set('NotificheSistema',true);msg.payload.content ='notifiche di sistema attivate';}\nelse {global.set('NotificheSistema',false);msg.payload.content = 'notifiche di sitema disattivate';}\nreturn msg;\n","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":1250,"y":1200,"wires":[["39f83657.5b0fd2"]]},{"id":"2d495163.537bc6","type":"telegram command","z":"9cf3a6d.1104258","name":"Abilita log rete","command":"/rete","description":"","registercommand":false,"language":"","scope":"default","bot":"cd6e647f.ad618","strict":false,"hasresponse":true,"useregex":false,"removeregexcommand":false,"outputs":2,"x":1020,"y":1320,"wires":[["68918b65.49f31c"],["c20728c.9d01a58"]]},{"id":"68918b65.49f31c","type":"function","z":"9cf3a6d.1104258","name":"Attiva?","func":"context.global.keyboard = {pending:true};\nvar opts = {reply_to_message_id: msg.payload.messageId, \n\treply_markup: JSON.stringify({keyboard: [ ['Attiva'], ['Disattiva']], 'resize_keyboard' : true,  'one_time_keyboard' : true })};\nmsg.payload.content = 'Attiva le notifiche per le risorse di rete?';\nmsg.payload.options = opts;\nreturn [ msg ];","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":1230,"y":1280,"wires":[["39f83657.5b0fd2"]]},{"id":"c20728c.9d01a58","type":"function","z":"9cf3a6d.1104258","name":"Quindi esegui","func":"context.global.keyboard = {pending:false};\nif(msg.payload.content === 'Attiva') {global.set('NotificheRete',true);msg.payload.content ='notifiche di rete attivate';}\nelse {global.set('NotificheRete',false);msg.payload.content = 'notifiche di rete disattivate';}\nreturn msg\n","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":1250,"y":1340,"wires":[["39f83657.5b0fd2"]]},{"id":"4d3926a7.cea2f","type":"debug","z":"9cf3a6d.1104258","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":1630,"y":1420,"wires":[]},{"id":"afae41ef.f7367","type":"inject","z":"9cf3a6d.1104258","name":"Debug","props":[{"p":"payload"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"true","payloadType":"bool","x":1630,"y":1340,"wires":[["3237249d.af7b1c"]]},{"id":"3237249d.af7b1c","type":"function","z":"9cf3a6d.1104258","name":"Output in debug delle variabili","func":"//Notifiche:global.get(\"Notifiche\"),\nmsg={\nNotificheSistema:global.get(\"NotificheSistema\"),\nNotificheRete:global.get(\"NotificheRete\"),\nNotificheDomotica:global.get(\"NotificheDomotica\")};\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":1670,"y":1380,"wires":[["4d3926a7.cea2f"]]},{"id":"94c2ceaf.756168","type":"inject","z":"9cf3a6d.1104258","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"false","payloadType":"bool","x":320,"y":1420,"wires":[["bb33a59a.cba3b"]]},{"id":"a1cffcc2.f1e39","type":"inject","z":"9cf3a6d.1104258","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"true","payloadType":"bool","x":350,"y":1600,"wires":[["bb33a59a.cba3b"]]},{"id":"cd6e647f.ad618","type":"telegram bot","botname":"DomoticaFrancoBot","usernames":"","chatids":"","baseapiurl":"","updatemode":"polling","pollinterval":"200","usesocks":false,"sockshost":"","socksport":"","socksusername":"","sockspassword":"","bothost":"","botpath":"","localbotport":"8443","publicbotport":"8443","privatekey":"","certificate":"","useselfsignedcertificate":false,"sslterminated":false,"verboselogging":false}]`

**Comment by Barabba11** [comment]
since now it didn't stop receiving messages (I?m testing with the echo function node), but I can't get rid from the keyboard, once I send command to rise it, The only way is to restart node itself (or whole NR) again

**Comment by windkh** [comment]
Hi @Barabba11 
I tested your flow (after removing the parts that I do not need) and it basically worked with my Telegram Desktop Client on Windows. The keyboard disappears correctly. So what client do you use?

![image](https://user-images.githubusercontent.com/10910767/215320607-94819ee5-8506-4daf-a5ca-e64eeb720a1e.png)

**Comment by windkh** [comment]
@Barabba11  Note that I disconnected your receiver node as I do not know what it does

**Comment by Barabba11** [comment]
thank you for testing it, I'm using raspberry 3B, with latest packages/NR/module, after updating (but keeping the 14.8, because the 15.0.1 was not available yet on palette, the good news is that the receiving node, that I use as an echo (I send "hello" and I see "hello" coming back, showing me system is online and promptly can notice me about everything, and this is its reason/meaning in the flow) now always work.. not like before it sopped working after one day (but transmit node always worked),
They keboard not disappeared. Now I updated with 15.0.1:
after process completed the keyboard in my telegram chat disappeared, I send a command and correctly appears, I select the answer, it keeps there.

**Comment by Barabba11** [comment]
I would correct the prevous post, I've restarted NR because the module wasn't updated wthout it, and I also deleted chat on prone and searched again for bot. With 15.0.1 situation hasn't changed, whrn I opened the chat the keayboard was there already, so mabe this is a bug of telegram android app? Keyboard keeps there, anyway the commands are working. SOmethign still in that chat which is unsatisfied
